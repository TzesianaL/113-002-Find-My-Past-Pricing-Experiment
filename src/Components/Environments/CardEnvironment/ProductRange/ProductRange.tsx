import React, { FC, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { addOutput, OutputTypes } from 'redux/reducers/output/outputSlice'
import { selectStoreItems } from 'redux/reducers/dataStore/dataStore.selectors'
import CardGroup from '../CardGroup/CardGroup'
import { StorePageStyle } from '../StorePage/StorePageStyle'

const ProductRange: FC = () => {
  const items = useSelector(selectStoreItems)

  const dispatch = useDispatch()
  useEffect(() => {
    dispatch(
      addOutput({
        key: OutputTypes.itemOrderShown,
        value: items.map((item) => item.id).toString(),
      })
    )
    const handleCardClick = () => {
      // setCardClick(true)
      console.log('ButtonClicked')
    }
    dispatch(
      addOutput({
        key: OutputTypes.numberOfItemsShown,
        value: String(items.length),
      })
    )
  }, [dispatch, items])

  return (
    <StorePageStyle>
      {/* <CardGroup
        items={items}
        hidePrice={false}
        handleCardClick={handleCardClick}
        purchaseButtonText="No purchase button"
      /> */}
    </StorePageStyle>
  )
}

export default ProductRange
