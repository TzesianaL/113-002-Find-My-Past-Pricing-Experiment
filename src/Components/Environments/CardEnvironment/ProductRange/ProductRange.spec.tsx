import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import ProductRange from './ProductRange'

const mockStore = configureStore([])

describe('<ProductRange />', () => {
  let store

  beforeEach(() => {
    store = mockStore({
      data: {
        dataLoaded: false,
        condition: -1,
        storeLogoSrc: '',
        storeName: 'Queen Carlotta`s undefined Store!',
        splash: {
          title: '',
          items: [],
        },
        storeItems: [],
        theme: {},
      },
    })
  })

  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductRange />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render component with minimum data requirements', () => {
    const el = render(
      <Provider store={store}>
        <ProductRange />
      </Provider>
    )

    expect(el).not.toBeNull()
  })
})
