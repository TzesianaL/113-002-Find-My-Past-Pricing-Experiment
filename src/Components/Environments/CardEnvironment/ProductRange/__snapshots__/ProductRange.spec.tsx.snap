// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRange /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 3em;
  background: #333333;
}

.c0 h2,
.c0 h1 {
  color: white;
}

.c0 .introText {
  font-size: 1.2em;
  max-width: 50em;
}

.c0 .footerText {
  font-size: 1.2em;
  max-width: 50em;
}

<div
  className="c0"
/>
`;
